volume/store/db.go

```diff
@@ -0,0 +1,79 @@
+package store
+
+import (
+	"encoding/json"
+
+	"github.com/boltdb/bolt"
+	"github.com/pkg/errors"
+)
+
+var volumeBucketName = []byte("volumes")
+
+type dbEntry struct {
+	Key   []byte
+	Value []byte
+}
+
+type volumeMetadata struct {
+	Name    string
+	Driver  string
```

Comment:
this struct has a new `Driver` field. Does this affect upgrade from 1.12 volumes?